src/lib/device-types/storage/s3.ts

```diff
@@ -110,6 +111,16 @@ export function fileExists(path: string): Promise<boolean> {
 	return getFileInfo(path)
 		.return(true)
 		.catch(err => {
+			// if we get an unauthorized error
+			// then we know that the file exists
+			// but has wrong permissions
+			if (err.statusCode === 403) {
+				captureException(
+					err,
+					`Got unathorized error while accessing path ${path}`,
+				);
+				return true;
```

Comment:
Might worth transforming the error to a specific PineJS one and let it propagate instead. I'd expect callers to invoke this function expecting the file they requested is also readable if it exists and in our case it's pretty much guaranteed that subsequent attempts to read it will fail (because requests will still go via the same AWS client with the same -- or none at all -- credentials). It's only used in a single place currently though, so I don't feel so strongly about this, it's up to you.

Reply:
As you can guess this was added to handle the case that an IGNORE file is added but it's not configured as public. With this the api will respect the IGNORE file instead of failing the whole device type.
@dfunckt Would you prefer
* to move the explicit 403 handling to `getIsIgnored()` instead?
* remove it completely and let the whole build promise get rejected, thus treating the build as invalid and omitting it? (This was the first case I tried to fix but it seems that removing it will have the same results)

Reply:
@thgreasi IMO the second option is the best -- same result with less code and no edge cases to handle 👍  If in the future we use `fileExists` elsewhere and don't really want it to fail like this, we should handle the errors at the call site and act appropriately (your first option).

Reply:
@dfunckt ended up removing the 403 handling as discussed. The api test cases seem happy with it :)